tests/console/pam.pm

```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This PM is to run a suite of tests about PAM, and gets a TAP
+#          format result. Four files(pam.sh, pam_test, pam_test.sh,
+#          system-default) are included. pam.sh is a "bats" script, and
+#          includes 25 tests for PAM. The other files are from the project
+#          "pam_test" which can be used to test a PAM stack for authentication
+#          and password change. The link: https://github.com/pbrezina/pam-test
+#   Steps:
+#       - add qa-head repo and install the tool "bats" from qa-head repo.
+#         Bats is a TAP-compliant testing framework for Bash.
+#       - download "pam_test" dir(automated tests of PAM) from "data" dir
+#         on openqa host if "prepare_test_data" perl module desn't run.
+#       - run the suite of tests in SUT.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils qw(is_leap is_sle is_tumbleweed is_opensuse);
+
+sub run {
+    select_console 'root-console';
+
+    my $qa_head_repo;
+    my $version = get_required_var('VERSION');
+    $qa_head_repo = "http://download.suse.de/ibs/QA:/Head/" . 'SLE-' . $version        if is_sle();
+    $qa_head_repo = "https://download.opensuse.org/repositories/devel:/openSUSE:/QA:/" if is_opensuse();
+    $qa_head_repo .= $version . '/openSUSE_' . $version if is_tumbleweed();
+    $qa_head_repo .= 'Leap:/' . substr($version, 0, index($version, '.')) . '/openSUSE_Leap_' . $version if is_leap();
```

Comment:
If the test won't be scheduled for openSUSE Leap or Tumbleweed, this can go away.

Reply:
OK, thank you for help me.

Reply:
well, the test *should* be scheduled for Tumbleweed at least but then you do not need to add the repo at all so just wrap the whole block in a `if (is_sle || is_leap) {`

Reply:
I see. pam-test is in openSUSE:Factory, so I don't need to add a additional repo for Tumbleweed, but for SLE and Leap, I need to add a additional repo, such as QA:Head for SLE and devel:openSUSE:QA:Leap for Leap.

Reply:
yes +1